File: ruby/tests/logger_test.rb
# frozen_string_literal: true

require_relative "test_helper"

class RedactTest < Minitest::Test
  def test_masks_sensitive_keys
    out = Pagou.redact({ "Authorization" => "Bearer abc", "token" => "pgct_123", "amount" => 4900 })
    assert_equal "[REDACTED]", out["Authorization"]
    assert_equal "[REDACTED]", out["token"]
    assert_equal 4900, out["amount"]
  end

  def test_masks_card_tokens_and_bearer_strings_inside_free_text
    assert_equal "charge with [REDACTED]", Pagou.redact("charge with pgct_secret123")
    assert_equal "header [REDACTED] here", Pagou.redact("header Bearer sk_live_xyz here")
  end

  def test_redacts_nested_structures
    out = Pagou.redact({ "buyer" => { "name" => "Ana", "document" => { "number" => "19100000000" } } })
    assert_equal "Ana", out["buyer"]["name"]
    assert_equal "[REDACTED]", out["buyer"]["document"]["number"]
  end

  def test_handles_circular_references
    obj = { "a" => 1 }
    obj["self"] = obj
    out = Pagou.redact(obj)
    assert_equal "[Circular]", out["self"]
  end
end

File: php/tests/LoggerTest.php
<?php

declare(strict_types=1);

namespace Pagou\Examples\Tests;

use Pagou\Examples\Logger;
use PHPUnit\Framework\TestCase;

final class LoggerTest extends TestCase
{
    public function testMasksSensitiveKeys(): void
    {
        $out = Logger::redact(['Authorization' => 'Bearer abc', 'token' => 'pgct_123', 'amount' => 4900]);
        self::assertSame('[REDACTED]', $out['Authorization']);
        self::assertSame('[REDACTED]', $out['token']);
        self::assertSame(4900, $out['amount']);
    }

    public function testMasksCardTokensAndBearerStringsInsideFreeText(): void
    {
        self::assertSame('charge with [REDACTED]', Logger::redact('charge with pgct_secret123'));
        self::assertSame('header [REDACTED] here', Logger::redact('header Bearer sk_live_xyz here'));
    }

    public function testRedactsNestedStructures(): void
    {
        $out = Logger::redact(['buyer' => ['name' => 'Ana', 'document' => ['number' => '19100000000']]]);
        self::assertSame('Ana', $out['buyer']['name']);
        self::assertSame('[REDACTED]', $out['buyer']['document']['number']);
    }
}
